torch/csrc/autograd/init.cpp

```diff
@@ -50,6 +50,22 @@ PyObject * THPAutograd_initExtension(PyObject *_unused)
   });
   m.def("_pop_range", []() { torch::autograd::profiler::popRange(); });
 
+  // used for testing and debugging
+  m.def("_register_callbacks", [](py::function start_fn, py::function end_fn, py::object extra_arg) {
+    start_fn.inc_ref(); //(?)
+    end_fn.inc_ref();
+    extra_arg.inc_ref();
+    torch::autograd::profiler::RecordFunctionRegistry::registerCallbacks(
```

Comment:
The formulation as two std::functions makes it hard for state to be passed from the entry of the recording block to the exit. It is not problem for the current use since it has a thread-local push/pop stack, but in other uses (e.g. if we unify this with the tracer) will have difficulty correlating a start_fn with an end_fn. I don't have a great solution for this though.